server/imports/core/connection/index.js

```diff
@@ -1,77 +1,154 @@
 import { Database, Logger, Error } from '/server/imports/modules';
 import ConnectionHelper from './helper';
 
 const fs = require('fs');
 const mongodbUrlParser = require('mongodb-url');
 const jsonParser = require('json5');
 
-const Connection = function () {
-};
+const Connection = function () {};
 
 const addSSLOptions = function (obj, result) {
   if (obj.rootCAFile) {
     result.sslValidate = true;
     result.sslCA = Buffer.from(obj.rootCAFile);
   }
   if (obj.certificateFile) result.sslCert = Buffer.from(obj.certificateFile);
-  if (obj.certificateKeyFile) result.sslKey = Buffer.from(obj.certificateKeyFile);
+  if (obj.certificateKeyFile) {
+    result.sslKey = Buffer.from(obj.certificateKeyFile);
+  }
   if (obj.passPhrase) result.sslPass = obj.passPhrase;
   if (obj.disableHostnameVerification) result.checkServerIdentity = false;
 };
 
 const setOptionsToConnectionFromParsedUrl = function (connection, parsedUrl) {
   if (parsedUrl.server_options) {
-    connection.options.connectionTimeout = (parsedUrl.server_options.socketOptions && parsedUrl.server_options.socketOptions.connectTimeoutMS)
-      ? parsedUrl.server_options.socketOptions.connectTimeoutMS : '';
-    connection.options.socketTimeout = (parsedUrl.server_options.socketOptions && parsedUrl.server_options.socketOptions.socketTimeoutMS)
-      ? parsedUrl.server_options.socketOptions.socketTimeoutMS : '';
+    connection.options.connectionTimeout =
+      parsedUrl.server_options.socketOptions &&
+      parsedUrl.server_options.socketOptions.connectTimeoutMS
+        ? parsedUrl.server_options.socketOptions.connectTimeoutMS
+        : '';
+    connection.options.socketTimeout =
+      parsedUrl.server_options.socketOptions &&
+      parsedUrl.server_options.socketOptions.socketTimeoutMS
+        ? parsedUrl.server_options.socketOptions.socketTimeoutMS
+        : '';
     connection.ssl.enabled = !!parsedUrl.server_options.ssl;
   }
 
-  connection.options.replicaSetName = (parsedUrl.rs_options && parsedUrl.rs_options.rs_name) ? parsedUrl.rs_options.rs_name : '';
+  connection.options.replicaSetName =
+    parsedUrl.rs_options && parsedUrl.rs_options.rs_name
+      ? parsedUrl.rs_options.rs_name
+      : '';
   connection.options.readPreference = parsedUrl.db_options.read_preference;
 };
 
 const setAuthToConnectionFromParsedUrl = function (connection, parsedUrl) {
-  connection.authenticationType = parsedUrl.db_options.authMechanism ? parsedUrl.db_options.authMechanism.toLowerCase().replace(new RegExp('-', 'g'), '_') : '';
-  if (connection.authenticationType) connection[connection.authenticationType] = {};
-  if (parsedUrl.db_options.gssapiServiceName && connection.authenticationType === 'gssapi') connection.gssapi.serviceName = parsedUrl.db_options.gssapiServiceName;
+  connection.authenticationType = parsedUrl.db_options.authMechanism
+    ? parsedUrl.db_options.authMechanism
+        .toLowerCase()
+        .replace(new RegExp('-', 'g'), '_')
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 6 spaces but found 8.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
      .replace(new RegExp('-', 'g'), '_')
```